docs/workers.md

```diff
@@ -467,10 +467,19 @@ the following regular expressions:
 
     ^/_matrix/client/(api/v1|r0|v3|unstable)/user_directory/search$
 
-When using this worker you must also set `update_user_directory: False` in the
+When using this worker you must also set `update_user_directory: false` in the
 shared configuration file to stop the main synapse running background
 jobs related to updating the user directory.
 
+Above endpoint is not *required* to be routed to this worker. By default,
+`update_user_directory` is set to `true`, which means the main process
+will handle updates. All workers configured with `client` can handle the above
+endpoint as long as either this worker or the main process are configured to
+handle it, and are online.
```

Comment:
Bleurgh, this is a bit annoying to document. Ideally we'd deprecate `synapse.app.user_dir` and replace it with an option in `generic_worker` (like for background tasks).

Reply:
There's some talk about this in #10339, I don't particularly know how to resolve this now.

Reply:
Given the scale of the discussion in #10339, I would document the current state now and get this piece of the puzzle merged. Then we can worry about from the transition from worker type -> generic worker option (https://github.com/matrix-org/synapse/issues/10441).

Reply:
So should I additionally document the `synapse.app.user_dir` worker type in `workers.md`?

Reply:
Actually, the worker type is already documented, i think that we shouldnt encode future intentions in the documentation, until we actually are working on it.